hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/ContainerReportHandler.java

```diff
@@ -102,8 +102,15 @@ public ContainerReportHandler(final NodeManager nodeManager,
   public void onMessage(final ContainerReportFromDatanode reportFromDatanode,
                         final EventPublisher publisher) {
 
-    final DatanodeDetails datanodeDetails =
+    final DatanodeDetails dnFromReport =
         reportFromDatanode.getDatanodeDetails();
+    DatanodeDetails datanodeDetails =
```

Comment:
When SCM does not have information about a datanode, it is usually the datanode is dead. So skip processing those delayed report makes sense to me .

Reply:
Xiaoyu, the key point here is we need the DatanodeDetails object from the nodeManager which has topology parent  correctly set.  The DatanodeDetails object parsed from container report always has no parent which will make the Replication Manager treat it as a datanode under /default-rack, further treat the container as under-replicated since cross rack requirement is not satisfied.

Reply:
That's a good catch. Thanks for the details.